modules/client/src/internal/constants.ts

```diff
@@ -121,3 +125,32 @@ export enum PreparationType {
   UPDATE = 2,
   UNINSTALLATION = 3,
 }
+
+export const UPDATE_PLUGIN_SIGNATURES: string[] = [
+  DAO__factory.createInterface().getFunction("grant")
+    .format("minimal"),
+  DAO__factory.createInterface().getFunction("revoke")
+    .format("minimal"),
+  PluginSetupProcessor__factory.createInterface().getFunction(
+    "applyUpdate",
+  ).format("minimal"),
+  DAO__factory.createInterface().getFunction(
+    "upgradeTo",
+  ).format("minimal"),
+  DAO__factory.createInterface()
+    .getFunction("upgradeToAndCall")
+    .format("minimal"),
+];
+
+export const PLUGIN_UPDATE_ACTION_PATTERN = [
+  ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+  ProposalActionTypes.APPLY_UPDATE,
+  ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+];
+export const PLUGIN_UPDATE_WITH_ROOT_ACTION_PATTERN = [
+  ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+  ProposalActionTypes.GRANT_ROOT_PERMISSION,
+  ProposalActionTypes.APPLY_UPDATE,
+  ProposalActionTypes.REVOKE_ROOT_PERMISSION,
+  ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+];
```

Comment:
I think it is ok to do this in a second iteration when we implement the rule-based solution you proposed @mathewmeconry. Since the SDK will produce these actions, we can assume that they have this order for now. Deviating from them is not necessary. The goal for now is that we can exclude trickery/malicious stuff.

Reply:
It should be pretty easy to add it here if you really want, but if ultimately we are going to scrap this I agree with @Michael-A-Heuer